### PR TITLE
Add support for yarn package manager

### LINKED ISSUE #30: Add Support for yarn Package Manager

### PR DESCRIPTION
**Pull request by shrvansudhakara**
## 🗒️ Description
Add support for Yarn package manager 

Closes #30 

### New Files
- `cmd/npm/yarn.go` - Yarn command handler with flow execution
- `packagemanager/yarn_test.go` - Comprehensive test suite for yarn (12 test cases)

### Modified Files
- `packagemanager/npm.go` - Added yarn configuration and dev flag support
- `internal/analytics/event.go` - Added yarn analytics tracking
- `main.go` - Registered yarn command
- `internal/alias/alias.go` - Added yarn to package managers list
- `go.mod` - Updated Go module dependencies
- `README.md` - Updated documentation with yarn support

### Yarn Command Support
- Implements `yarn add <package>` for package installation
- Implements `yarn install` for manifest-based installation
- Supports version specifications: `yarn add package@version`
- Handles dev dependencies with `--dev` and `-D` flags
- Supports global package installation: `yarn global add <package>`

### Key Differences from npm/pnpm
- Uses `add` command instead of `install` for adding packages
- Uses `--dev`/`-D` flag instead of `--save-dev` for dev dependencies
- Follows the same guard flow as other npm ecosystem package managers

### Configuration
```go
DefaultYarnPackageManagerConfig() {
    InstallCommands: []string{"install", "add"},
    CommandName:     "yarn",
}
```

### README.md Changes
- Updated package managers table: yarn status changed from `🚧 Planned` to `✅ Active`
- Added `pmg yarn add <package>` to command examples
- Added `yarn add <package-name>` to usage examples
- Added `yarn install # Uses yarn.lock` to lockfile examples

### Test Coverage
All tests passing (12 new yarn tests + all existing tests):
- Install single package
- Install package with specific version
- Install development packages (`--dev` and `-D` flags)
- Manifest-based installation (`yarn install`)
- Multiple package installations
- Skip non-installation commands
- Handle intermediate flags
- Global package installation
- Edge cases (no packages, mixed args, etc.)

### Run Tests
```bash
# Run yarn-specific tests
go test ./packagemanager -v -run TestYarnParseCommand

# Run all tests
go test ./... -v
```

### Verification
```bash
# Build the project
go build

# Check yarn command is registered
./pmg --help | grep yarn

# Setup aliases
./pmg setup install

# Verify yarn alias
cat ~/.pmg.rc | grep yarn
```

## 📋 Checklist
- [x] Code follows project conventions
- [x] All tests passing
- [x] New tests added for yarn support
- [x] Command registered in main.go
- [x] Analytics tracking added
- [x] Alias setup updated
- [x] Documentation updated in README
- [x] Commit signed with `-s` flag
- [x] No breaking changes to existing functionality

## 📸 Screenshots

**Help Output:**
pmg-ss-1
```
pmg> ./pmg --help

█▀█ █▀▄▀█ █▀▀   From SafeDep
█▀▀ █░▀░█ █▄█   version: v0.1.0+dirty commit:

Usage:
  pmg [flags]
  pmg [command]
...
```

**Alias Setup:**
pmg-ss-2
```
pmg> ./pmg setup install

█▀█ █▀▄▀█ █▀▀   From SafeDep
█▀▀ █░▀░█ █▄█   version: v0.1.0+dirty commit:

✅ PMG aliases installed successfully!
📁 Created: C:\Users\Shrvan\.pmg.rc
💡 Restart your terminal or source your shell to use the new aliases
```

## 📚 Related

- Closes #30
- Follows the same pattern as npm, pnpm, and bun implementations
- Part of expanding npm ecosystem support

## 🤝 Reviewers
@abhisek, @KunalSin9h, @Sahilb315  

---

**Note:** This PR maintains backward compatibility and doesn't modify any existing package manager functionality. All existing tests remain passing.

### CODE REVIEW

**Comment by shrvansudhakara** [comment]
@abhisek, @KunalSin9h Please review the PR.

File: packagemanager/npm.go [hunk 1]
```diff
@@ -36,6 +36,13 @@ func DefaultBunPackageManagerConfig() NpmPackageManagerConfig {
 	}
 }
 
+func DefaultYarnPackageManagerConfig() NpmPackageManagerConfig {
```
Comment (Sahilb315) [inline packagemanager/npm.go:39]: @shrvansudhakara 
As per `yarn` documentation, `yarn` itself is a install command (aliased to `yarn install`): https://yarnpkg.com/getting-started/usage

Please add support for `yarn` cmd as manifest based installation

File: packagemanager/npm.go [hunk 2]
```diff
@@ -36,6 +36,13 @@ func DefaultBunPackageManagerConfig() NpmPackageManagerConfig {
...
+	return NpmPackageManagerConfig{
+		InstallCommands: []string{"install", "add"},
```
Comment (Sahilb315) [inline packagemanager/npm.go]: `yarn install <package>` is not a valid command for installing package(s). `yarn install` is used to install all dependencies for a project.
Please check: https://classic.yarnpkg.com/en/docs/cli/install

Eg: 
image

File: extractor/npm.go [hunk 3]
```diff
@@ -59,3 +59,21 @@ func (n *BunExtractor) GetPackageManager() PackageManagerName {
 func (n *BunExtractor) Extract(lockfilePath, scanDir string) ([]*packagev1.PackageVersion, error) {
 	return parseLockfile(lockfilePath, scanDir, n.GetEcosystem())
 }
+
+type YarnExtractor struct{}
```
Comment (Sahilb315) [inline extractor/npm.go:63]: @shrvansudhakara 

You will have to add the mapping for `yarn.lock` in https://github.com/safedep/pmg/blob/main/extractor/common.go#L21

After changes, please do test out if both (package & manifest based) installation command are working

File: packagemanager/npm.go [hunk 4]
```diff
@@ -115,6 +132,13 @@ func (npm *npmPackageManager) ParseCommand(args []string) (*ParsedCommand, error
 		isManifestInstall = true
 	}
 
+	// Yarn-specific validation: yarn install does not accept package names
+	if npm.Config.CommandName == "yarn" && args[installCmdIndex] == "install" && len(packages) > 0 {
+		return &ParsedCommand{
+			Command: command,
+		}, nil
+	}
+
```
Comment (Sahilb315) [inline packagemanager/npm.go:141]: @shrvansudhakara 
Can you add a test case to verify this? `yarn install express` returns command with no `InstallTargets` & is not a `ManifestInstall`

File: packagemanager/npm.go [hunk 5]
```diff
@@ -36,6 +36,13 @@ func DefaultBunPackageManagerConfig() NpmPackageManagerConfig {
 	}
 }
 
+func DefaultYarnPackageManagerConfig() NpmPackageManagerConfig {
+	return NpmPackageManagerConfig{
+		InstallCommands: []string{"install", "add", ""},
```
Comment (Sahilb315) [inline packagemanager/npm.go:41]: @shrvansudhakara 
Why do we require `""` as an `InstallCommand`?

Comment (shrvansudhakara) [reply packagemanager/npm.go:41]: @Sahilb315 

The `""`  in `InstallCommands` is needed to handle the bare `yarn` command.

When a user runs just `yarn`, the args become empty after parsing. The empty string `""` in `InstallCommands` allows the code to recognize this as a valid install command and trigger manifest-based installation.

This is as per [yarn documentation](https://yarnpkg.com/getting-started/usage) where `yarn` is an alias for `yarn install`.

**Review by abhisek — APPROVED** [review]
